google-cloud-logging/src/main/java/com/google/cloud/logging/Exclusion.java

```diff
@@ -0,0 +1,259 @@
+      return new Exclusion(this);
+    }
+  }
+
+  Exclusion(Builder builder) {
+    this.name = checkNotNull(builder.name);
+    this.description = builder.description;
+    this.filter = checkNotNull(builder.filter);
+    this.disabled = builder.disabled;
+    this.createTime = builder.createTime;
+    this.updateTime = builder.updateTime;
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("name", name)
+        .add("description", description)
+        .add("filter", filter)
+        .add("disabled", disabled)
+        .add("createTime", createTime)
+        .add("updateTime", updateTime)
+        .toString();
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Exclusion exclusion = (Exclusion) o;
+    return disabled == exclusion.disabled
+        && Objects.equals(name, exclusion.name)
+        && Objects.equals(description, exclusion.description)
+        && Objects.equals(filter, exclusion.filter)
+        && Objects.equals(createTime, exclusion.createTime)
+        && Objects.equals(updateTime, exclusion.updateTime);
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(name, description, filter, disabled, createTime, updateTime);
+  }
+
+  /** Returns the name of log exclusion. */
+  public String getName() {
+    return name;
+  }
+
+  /** Returns an optional description of an exclusion. Used for documentation purpose. */
+  public String getDescription() {
+    return description;
+  }
+
+  /**
+   * Returns an advanced logs filter. Example: {@code resource.type=gcs_bucket severity<ERROR
+   * sample(insertId, 0.99)}.
+   *
+   * @see <a href="https://cloud.google.com/logging/docs/view/advanced-queries">Advanced Log
+   *     Filters</a>
+   */
+  public String getFilter() {
+    return filter;
+  }
+
+  /** If set to True, then this exclusion is disabled and it does not exclude any log entries. */
+  public boolean isDisabled() {
+    return disabled;
+  }
+
+  /** Returns the creation timestamp of the exclusion. */
+  public Timestamp getCreateTime() {
+    return createTime;
+  }
+
+  /** Returns the last update timestamp of the exclusion. */
+  public Timestamp getUpdateTime() {
+    return updateTime;
+  }
+
+  /** Returns a builder for this {@code Exclusion} object. */
+  public Builder toBuilder() {
+    return new Builder(this);
+  }
+
+  /**
+   * Returns a builder for {@code Exclusion} objects given the name of the exclusion and its filter.
+   */
+  public static Builder newBuilder(String name, String filter) {
+    return new Builder(name, filter);
+  }
+
+  /** Creates a {@code Exclusion} object given the name of the exclusion and its filter. */
+  public static Exclusion of(String name, String filter) {
+    return new Builder(name, filter).build();
+  }
+
+  LogExclusion toPb() {
```

Comment:
Haven't we discussed that we should go with `toProtobuf()` instead of `toPb()`... was there any particular reason to go with `toPb()`?

Reply:
done